core/tee/tee_svc_storage.c

```diff
@@ -307,6 +307,12 @@ static TEE_Result tee_svc_storage_read_head(struct tee_ta_session *sess,
 
 	fops = o->pobj->fops;
 
+	o->pobj->head = malloc(sizeof(struct tee_svc_storage_head));
```

Comment:
If B has written data after A, wouldn't o->info.dataSize be updated with the new size? There could a race condition unless we have a mutex that protects o.